src/TopicCard.jsx

```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import codingTopicImage from "./images/coding-topic.jpg";
+import footballTopicImage from "./images/football-topic.jpg";
+import cookingTopicImage from "./images/cooking-topic.jpg";
+import "./App.css";
+
+const TopicCard = ({ topic }) => {
+  let topicImage = "path/to/placeholder/image.jpg";
+
+  if (topic.slug === "coding") {
+    topicImage = codingTopicImage;
+  }
+  if (topic.slug === "football") {
+    topicImage = footballTopicImage;
+  }
+  if (topic.slug === "cooking") {
+    topicImage = cookingTopicImage;
+  }
```

Comment:
This is not dynamic and fails to work for any future topics that may be added but have not been accounted for here.